frontend/src/components/resourceMap/useQueryParamsState.tsx

```diff
@@ -0,0 +1,57 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useLocation } from 'react-router';
+
+type UseQueryParamsStateReturnType<T> = [T, Dispatch<SetStateAction<T>>];
+export function useQueryParamsState<T extends string | undefined>(
+  param: string,
+  initialState: T
+): UseQueryParamsStateReturnType<T> {
+  const location = useLocation();
+
+  // State for managing the value derived from the query parameter
+  const [value, setValue] = useState<T>(() => {
+    if (typeof window === 'undefined') return initialState;
+
+    // Parse query parameter value from the URL
+    const { search } = window.location;
+    const searchParams = new URLSearchParams(search);
```

Comment:
Since we're using react-router it would be good to use react-router functions for getting/setting.

Reply:
yeah this hook is going around react-router, I'm going to pretty much rewrite this

Reply:
done